diff --git a/src/ucp/core/ucp_device.c b/src/ucp/core/ucp_device.c
index 71d2895..5fb965a 100644
--- a/src/ucp/core/ucp_device.c
+++ b/src/ucp/core/ucp_device.c
@@ -398,7 +398,8 @@ static ucs_status_t ucp_device_local_mem_list_params_check(
             return UCS_ERR_INVALID_PARAM;
         }
 
-        if (memh->sys_dev != local_sys_dev) {
+        if ((memh->sys_dev != local_sys_dev) &&
+            (memh->mem_type != UCS_MEMORY_TYPE_HOST)) {
             ucs_error("mismatched local sys_dev for element=%zu", i);
             return UCS_ERR_UNSUPPORTED;
         }
